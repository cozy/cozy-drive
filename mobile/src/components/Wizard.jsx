import React, { Component } from 'react'

export default class Wizard extends Component {
  constructor (props) {
    super(props)

    this.nextStep = this.nextStep.bind(this)
    this.previousStep = this.previousStep.bind(this)

    this.state = {
      step: 0
    }
  }

  nextStep () {
    this.setState({ step: this.state.step + 1 })
  }

  previousStep () {
    if (this.state.step === 0) return
    this.setState({ step: this.state.step - 1 })
  }

  render () {
    const Step = this.props.steps[this.state.step]
    if (!Step) {
      if (this.props.location.state && this.props.location.state.nextPathname) {
        this.props.router.replace(this.props.location.state.nextPathname)
      } else {
        this.props.router.replace('/')
      }
    }
    return <Step nextStep={this.nextStep} previousStep={this.previousStep} {...this.props} />
  }
}
