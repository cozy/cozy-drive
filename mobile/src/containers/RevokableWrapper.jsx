/* global cozy */

import React, { Component } from 'react'
import { connect } from 'react-redux'
import Modal from 'cozy-ui/react/Modal'
import { translate } from '../../../src/lib/I18n'
import { resetClient } from '../lib/cozy-helper'
import { unrevokeClient } from '../actions/authorization'
import { registerDevice } from '../actions/settings'

class RevokableWrapper extends Component {
  logout () {
    resetClient()
    this.props.unrevokeClient()
    this.props.router.replace({
      pathname: '/onboarding',
      state: { nextPathname: this.props.location.pathname }
    })
  }

  loginagain () {
    cozy.client._storage.clear()
    this.props.registerDevice()
  }

  render () {
    const { children, t } = this.props
    if (this.props.revoked) {
      return (
        <div>
          <Modal
            title={t('mobile.revoked.title')}
            description={t('mobile.revoked.description')}
            secondaryText={t('mobile.revoked.logout')}
            secondaryAction={() => { this.logout() }}
            primaryText={t('mobile.revoked.loginagain')}
            primaryAction={() => { this.loginagain() }}
            withCross={false}
          />
          {children}
        </div>
      )
    } else {
      return children
    }
  }
}

const mapDispatchToProps = (dispatch) => ({
  unrevokeClient: () => dispatch(unrevokeClient()),
  registerDevice: () => dispatch(registerDevice()).then(() => { dispatch(unrevokeClient()) })
})

const mapStateToProps = (state) => ({
  revoked: state.mobile.authorization.revoked
})

export default connect(mapStateToProps, mapDispatchToProps)(translate()(RevokableWrapper))
