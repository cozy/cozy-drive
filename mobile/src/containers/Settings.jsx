import React from 'react'
import { connect } from 'react-redux'
import { translate } from '../../../src/lib/I18n'

import styles from '../styles/settings'

const SubCategory = ({id, label, value}) => (
  <div className={styles['settings__subcategory']}>
    <h4 className={styles['settings__subcategory-title']}>{label}</h4>
    <p id={id} className={styles['settings__subcategory-item']}>{value}</p>
  </div>
)

export const Settings = ({t, version, serverUrl}) => (
  <div>
    <div className={styles['fil-content-row']} />
    <div className={styles['settings']}>
      <h3 className={styles['settings__category-title']}>{t('mobile.settings.about')}</h3>
      <SubCategory id={'serverUrl'} label={t('mobile.settings.account')} value={<a href={serverUrl}>{serverUrl}</a>} />
      <SubCategory id={'version'} label={t('mobile.settings.app_version')} value={version} />
    </div>
  </div>
)

const mapStateToProps = (state, ownProps) => ({
  version: window.navigator.appInfo ? window.navigator.appInfo.version : 'dev',
  serverUrl: state.mobile.serverUrl
})

export default translate()(connect(mapStateToProps)(Settings))
