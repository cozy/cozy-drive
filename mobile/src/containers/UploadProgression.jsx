import { connect } from 'react-redux'
import UploadProgression from '../components/UploadProgression'

const mapStateToProps = (state) => ({
  message: state.mobile.mediaBackup.currentUpload ? state.mobile.mediaBackup.currentUpload.message : undefined,
  messageData: state.mobile.mediaBackup.currentUpload ? state.mobile.mediaBackup.currentUpload.messageData : undefined
})

export default connect(mapStateToProps)(UploadProgression)
