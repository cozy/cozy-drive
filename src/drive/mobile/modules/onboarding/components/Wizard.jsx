import React, { Component } from 'react'
import Breadcrumb from './Breadcrumb'

export default class Wizard extends Component {
  constructor(props) {
    super(props)

    this.nextStep = this.nextStep.bind(this)
    this.previousStep = this.previousStep.bind(this)

    this.state = {
      step: 0
    }
  }

  nextStep() {
    if (this.state.step >= this.props.steps.length - 1) {
      this.props.onComplete()
      return
    }
    this.setState({ step: this.state.step + 1 })
  }

  previousStep() {
    if (this.state.step === 0) return
    this.setState({ step: this.state.step - 1 })
  }

  render() {
    const Step = this.props.steps[this.state.step]
    const Breadcrumbs = (
      <Breadcrumb
        currentStep={this.state.step + 1}
        totalSteps={this.props.steps.length}
      />
    )

    return (
      <Step
        nextStep={this.nextStep}
        previousStep={this.previousStep}
        breadcrumbs={Breadcrumbs}
        {...this.props}
      />
    )
  }
}
