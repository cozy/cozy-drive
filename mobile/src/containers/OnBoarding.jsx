import React from 'react'
import classnames from 'classnames'
import { connect } from 'react-redux'
import { translate } from '../../../src/lib/I18n'

import Wizard from '../components/Wizard'

import styles from '../styles/onboarding'
import { registerDevice, SET_URL } from '../actions'

import logo from '../../res/icon.png'

export const SelectServer = ({selectServer, t, updateServerUrl, serverUrl, error}) =>
(
  <div className={classnames(styles['wizard'])}>
    <div className={classnames(styles['wizard-main'])}>
      <p>{t('mobile.onboarding.server_selection.cozy_address')}</p>
      <input type='url' placeholder={t('mobile.onboarding.server_selection.cozy_address_placeholder')} onChange={updateServerUrl} value={serverUrl} />
      <p>{t('mobile.onboarding.server_selection.description')}</p>
      {error && <p style={{color: 'red'}}>{t(error)}</p>}
    </div>
    <button role='button' className={classnames('coz-btn coz-btn--regular', styles['wizard-button'])} onClick={selectServer}>{t('mobile.onboarding.server_selection.button')}</button>
  </div>
)

const mapDispatchToProps = (dispatch, ownProps) => ({
  selectServer: () => {
    const { router, location } = ownProps
    dispatch(registerDevice(router, location))
  },
  updateServerUrl: (e) => {
    const serverUrl = e.target.value
    dispatch({ type: SET_URL, url: serverUrl })
  }
})

const mapStateToProps = (state) => {
  return ({
    serverUrl: state.mobile.serverUrl,
    error: state.mobile.error
  })
}
const ConnectedSelectServer = connect(mapStateToProps, mapDispatchToProps)(SelectServer)

export const Welcome = ({ nextStep, t }) =>
(
  <div className={classnames(styles['wizard'], styles['welcome'])}>
    <div className={classnames(styles['wizard-main'])}>
      <img src={logo} alt='logo' />
      <h1>{t('mobile.onboarding.welcome.title')}</h1>
    </div>
    <button role='button' className={classnames('coz-btn coz-btn--regular', styles['wizard-button'])} onClick={nextStep}>{t('mobile.onboarding.welcome.button')}</button>
  </div>
)

const OnBoarding = (props) => {
  const steps = [
    Welcome,
    ConnectedSelectServer
  ]
  return <Wizard steps={steps} {...props} />
}

export default translate()(OnBoarding)
