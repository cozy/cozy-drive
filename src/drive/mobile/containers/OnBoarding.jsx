import React, { Component } from 'react'

import Wizard from '../components/Wizard'

import Files from './onboarding/Files'
import BackupPhotosVideos from './onboarding/BackupPhotosVideos'
import BackupContacts from './onboarding/BackupContacts'
import Analytics from './onboarding/Analytics'

import { initBar } from 'drive/mobile/lib/cozy-helper'

export default class OnBoarding extends Component {
  onboardingSteps = [Files, BackupPhotosVideos, BackupContacts, Analytics]

  redirectToApp() {
    initBar(this.context.client)
    this.props.router.replace('/')
  }

  render() {
    return (
      <Wizard
        steps={this.onboardingSteps}
        onComplete={() => this.redirectToApp()}
      />
    )
  }
}
