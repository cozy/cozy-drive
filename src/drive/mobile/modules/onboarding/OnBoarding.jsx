import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { initBar } from 'drive/mobile/lib/cozy-helper'

import Wizard from './components/Wizard'
import Files from './components/Files'
import BackupPhotosVideos from './components/BackupPhotosVideos'
import Analytics from './components/Analytics'

export default class OnBoarding extends Component {
  static contextTypes = {
    client: PropTypes.object.isRequired
  }
  onboardingSteps = [Files, BackupPhotosVideos, Analytics]

  async redirectToApp() {
    await initBar(this.context.client)
    this.props.router.replace('/')
  }

  render() {
    return (
      <Wizard
        steps={this.onboardingSteps}
        onComplete={() => this.redirectToApp()}
      />
    )
  }
}
