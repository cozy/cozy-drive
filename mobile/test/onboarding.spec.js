'use strict'

import React from 'react'
import renderer from 'react-test-renderer'

import Wizard from '../../mobile/src/components/Wizard'
import { Welcome, SelectServer } from '../../mobile/src/containers/OnBoarding'

describe('Onboarding', () => {
  it('should render different components', () => {
    const steps = [
      Welcome,
      SelectServer
    ]
    const component = renderer.create(
      <Wizard steps={steps} t={() => {}} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.children[1].props.onClick()
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the welcome screen', () => {
    const component = renderer.create(<Welcome t={() => {}} />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the SelectServer screen', () => {
    const component = renderer.create(<SelectServer t={() => {}} />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
