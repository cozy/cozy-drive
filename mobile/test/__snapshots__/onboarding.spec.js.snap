exports[`Onboarding should render different components 1`] = `
<div
  className="wizard welcome">
  <div
    className="wizard-main">
    <img
      alt="logo"
      src="test-file-stub" />
    <h1 />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={[Function]}
    role="button" />
</div>
`;

exports[`Onboarding should render different components 2`] = `
<div
  className="wizard">
  <div
    className="wizard-main">
    <p />
    <input
      onChange={undefined}
      placeholder={undefined}
      type="url"
      value={undefined} />
    <p />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;

exports[`Onboarding should render the SelectServer screen 1`] = `
<div
  className="wizard">
  <div
    className="wizard-main">
    <p />
    <input
      onChange={undefined}
      placeholder={undefined}
      type="url"
      value={undefined} />
    <p />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;

exports[`Onboarding should render the welcome screen 1`] = `
<div
  className="wizard welcome">
  <div
    className="wizard-main">
    <img
      alt="logo"
      src="test-file-stub" />
    <h1 />
  </div>
  <button
    className="coz-btn coz-btn--regular wizard-button"
    onClick={undefined}
    role="button" />
</div>
`;
